### ISSUE TITLE
Diamond v.2.1.6 not recognizing input .fasta files

### ISSUE THREAD

**Issue by BirdmanRidesAgain** [bbuchfink/diamond#705]
Hi, 
I'm trying to run Diamond v.2.1.6, installed via the bioconda distribution, on a set of fasta files, using the following code:

```
diamond blastx --query HOU_ASM.purged.fa \
--db /path/to/reference_proteomes/ \
outfmt 6 qseqid staxids bitscore qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore \
--sensitive \
--max-target-seqs 1 \
--evalue 1e-25 \
--threads 10 > Tridacna_squamosina_flye_racon.diamond.blastx.out
```

However, every input fasta I've tried returns the same error:

`Error: Error detecting input file format. First line must begin with '>' (FASTA) or '@' (FASTQ)`

The fasta work in blastn, but I can't seem to get them to run here, and running ```head``` confirms that they _do_ begin with a '>'. I tried running dos2unix on them, on the off chance that there were some newlines or something interfering, but that didn't work.
Any guidance you have would be appreciated.

**Comment by bbuchfink** [comment]
I can't reproduce the problem. If you send me the query file in question I can look further.

**Comment by BirdmanRidesAgain** [comment]
Of course. I’ve subsampled the fasta to one to decrease the size enough to send via email, but it does still reproduce the error on my end.
However, if you can’t reproduce it, I assume there might be something wrong with the installation and/or distribution on my end. I’ll try and reinstall via bioconda to see if that sorts it out.
￼

**Comment by LR-2021** [comment]
Hi, 
I am getting the same issue. I have tested several versions. Have you solved the problem?

Thanks

**Comment by bbuchfink** [comment]
> Hi, I am getting the same issue. I have tested several versions. Have you solved the problem?
> 
> Thanks

What is your operating system and how did you download diamond? Can you send me your query file?

**Comment by NantiaL** [comment]
Hey, same happens in version 2.0.11. I will send you my query file.

**Comment by LR-2021** [comment]
Hi,

I haven't answered yet because it seems I have solved the problem. However, I'm still waiting to check the blastx results, which have been pending for the last 4 days. There was another issue, #726, which might have been related, but I cannot confirm it completely yet.

Following @bbuchfink's suggestion, I tested diamond with test data, and it worked perfectly. Then, I checked the headers (see #726) and confirmed that @bbuchfink was correct; I was somehow missing the correct header line in the -taxonmap file.

After that, I ran diamond without any problems, but I am still waiting for the results. Hopefully, it worked.

**Comment by hj1994412** [comment]
Subject: Error: Error detecting input file format with blastx despite seemingly valid FASTA input
Hi DIAMOND developers,

I am encountering a persistent error while using diamond blastx for annotating viral contigs. The program reports that it cannot detect the input file format, even though my input files appear to be in the correct FASTA format.

I would be very grateful for any guidance on how to resolve this.

Command Used

I am running the following command inside a bash for loop (where ${i} is a variable representing the cluster name):
`diamond blastx --query estuary.contig/metaT/cluster2/${i}.metaT.cluster2.fasta \
--db /mnt/md0/database/RdRp-scan/RdRp-scan_0.90.dmnd \
--min-orf 600 --evalue 1e-5 --very-sensitive \
--outfmt 6 qtitle stitle qstrand qseqid sseqid pident length mismatch gapopen qstart qend sstart send evalue bitscore \
--threads 10 --max-target-seqs 1 --quiet \
--out 06.RdRp/${i}.metaT.cluster2.diamond`

Error Message

For each file processed in the loop, I receive the following error:
Error: Error detecting input file format. First line must begin with '>' (FASTA) or '@' (FASTQ).

Troubleshooting Steps Taken

Based on the error message, the issue seems to point to the --query file. I have already checked the following:

File Format: I have inspected the input files (e.g., estuary.contig/metaT/cluster2/some_cluster_name.metaT.cluster2.fasta). They are not empty, and the first line correctly begins with a > character, as confirmed with head -n 1 <filename>.

File Content: The files appear to be standard, multi-line FASTA files. There are no obvious signs of corruption or unusual characters at the beginning of the file.

Shell Variable: The bash variable ${i} is expanding correctly and points to an existing file for each iteration of the loop.

System Information
DIAMOND Version: 2.1.11
Ubuntu 20 LTS
Installation Method: Conda

**Comment by bbuchfink** [comment]
@hj1994412 Could you send me one of your query files (buchfink@gmail.com)?

**Comment by hj1994412** [comment]
> [@hj1994412](https://github.com/hj1994412) Could you send me one of your query files ([buchfink@gmail.com](mailto:buchfink@gmail.com))?

Thank you. I have already resolved this issue—it wasn't a problem with the query files, but rather with the dmnd database. After redownloading the database, it now runs normally.

**Comment by bbuchfink** [comment]
@nickp60 thanks for figuring that out, I fixed the error message to be clearer in the latest version.
